### PR TITLE
[FrameworkBundle] Fix compatibility with symfony/security-core 6.x (deps=high tests)

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 5.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Complete diff of this PR must be removed when upmerging to 6.0.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Controller/AbstractControllerTest.php [hunk 1]
```diff
@@ -148,6 +153,11 @@ public function testGetUser()
 
     public function testGetUserAnonymousUserConvertedToNull()
     {
+        // @deprecated since Symfony 5.4
+        if (!class_exists(AnonymousToken::class)) {
+            $this->markTestSkipped('This test requires "symfony/security-core" <6.0.');
```
Comment (wouterj) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Controller/AbstractControllerTest.php:158]: This test is irrelevant in 6.0, as anonymous tokens are removed and anonymous sessions are now null anyways (so no conversion is needed in `AbstractController::getUser()`).

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Tests/Controller/AbstractControllerTest.php [hunk 2]
```diff
@@ -138,7 +138,12 @@ public function testForward()
     public function testGetUser()
     {
         $user = new InMemoryUser('user', 'pass');
-        $token = new UsernamePasswordToken($user, 'pass', 'default', ['ROLE_USER']);
+        if (method_exists(UsernamePasswordToken::class, 'setAuthenticated')) {
+            // @deprecated since Symfony 5.4
+            $token = new UsernamePasswordToken($user, 'pass', 'default', ['ROLE_USER']);
+        } else {
+            $token = new UsernamePasswordToken($user, 'default', ['ROLE_USER']);
+        }
```
Comment (wouterj) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Controller/AbstractControllerTest.php:146]: Expected BC break, correct deprecations are triggered on 5.4.

File: src/Symfony/Bundle/FrameworkBundle/KernelBrowser.php [hunk 3]
```diff
@@ -123,7 +123,10 @@ public function loginUser(object $user, string $firewallContext = 'main'): self
         }
 
         $token = new TestBrowserToken($user->getRoles(), $user, $firewallContext);
-        $token->setAuthenticated(true);
+        // @deprecated since Symfony 5.4
+        if (method_exists($token, 'setAuthenticated')) {
+            $token->setAuthenticated(true);
+        }
```
Comment (wouterj) [inline src/Symfony/Bundle/FrameworkBundle/KernelBrowser.php:129]: The real bugfix of this PR, this adds support for security-core 6.0 in FrameworkBundle.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @wouterj.

**Pull request merged**